RecoTauTag/RecoTau/src/PFTauPrimaryVertexProducerBase.cc

```diff
@@ -204,6 +204,14 @@ void PFTauPrimaryVertexProducerBase::produce(edm::Event& iEvent, const edm::Even
           } else {
             transVtx = avf.vertex(transTracks, *beamSpot);
           }
+          if (!transVtx.isValid()) {
+            fitOK = false;
+          } else {
+            //MB: protect against rare cases when transVtx is valid but its position is ill-defined
+            reco::Vertex::Point pvPos(transVtx.position());
+            if (!std::isfinite(pvPos.z()))  //MB: it is enough to check one coordinate (?)
```

Comment:
is `transVtx.position().z()` enough? The explicit conversion to another type is not particularly clearly necessary.

Reply:
Yes, right; I'll simplify the code.